core/sql/optimizer/NATable.cpp

```diff
@@ -5481,7 +5481,9 @@ NABoolean createNAFileSets(TrafDesc * table_desc       /*IN*/,
 
   if(postCreateNATableWarnings != preCreateNATableWarnings)
 	  tableConstructionHadWarnings_=TRUE;
-  const char *lobHdfsServer = CmpCommon::getDefaultString(LOB_HDFS_SERVER);
+  char  lobHdfsServer[256] ; // max length determined by dfs.namenode.fs-limits.max-component-length(255)
+  memset(lobHdfsServer,0,256);
+  strncpy(lobHdfsServer,CmpCommon::getDefaultString(LOB_HDFS_SERVER), strlen(CmpCommon::getDefaultString(LOB_HDFS_SERVER)));
```

Comment:
Doesn't that make the problem worse? If the string is larger than 256 bytes, it will overrun the lobHdfsServer buffer? Should the length passed in be MINOF(strlen(...), 256)?

Also, I use sizeof(lobHdfsServer) instead of 256 in these cases, but others have disagreed with me on that.

Also, shouldn't there be a global constant for the max. length of a server?